File: deneme/linked_list/linkedlist.py
class Node:
    def __init__(self,data) -> None:
        self.data = data
        self.next = None
        
class LinkedList:
    def __init__(self) -> None:
        self.head = None
        
    def append(self,data):
        new_node = Node(data)
        if self.head is None:
            self.head = new_node
            return
        
        last = self.head
        while last.next:
            last = last.next
        
        last.next = new_node
        
    def print_list(self):
        current = self.head
        while current:
            print(current.data, end = "->")
            current = current.next
        print("None")
    
    def delete_node(self,key):
        temp = self.head
        if temp is not None:
            if temp.data == key:
                self.head = temp.next
                temp = None
                return
            
        while temp is not None:
            if temp.data == key:
                break
            
            prev = temp
            temp = temp.next
        
        prev.next = temp.next
        temp = None
        
        if temp is not None:
            return
        

list  = LinkedList()
list.append(1)
list.append(2)
list.append(3)
list.append(4)

print("Linked List : ")
list.print_list()

list.delete_node(2)
list.print_list()            
            

File: deneme/linked_list/linkedList.cpp
#include <iostream>
using namespace std;

class Node{
public:
    int data; // tuttuğumuz elemanların değeri
    Node* next;

    Node(int data){
        this->data = data; 
        this->next = nullptr;
    }
};

class LinkedList{
public:
    Node* head; //başlangıç node'unu tanımlamak.

    LinkedList(){
        head = nullptr;
    }

    void append(int data1){
        Node* new_node = new Node(data1);
        if(head == nullptr){
            head = new_node;
            return;
        }
        Node* last = head; //son düğümü bulma işlemi
        while(last->next != nullptr){
            last = last->next;
        }
        last->next = new_node;
    }

    void printList(){
        Node* current = head;
        while(current != nullptr){
            cout << current->data << "->";
            current = current->next;
        }
        cout << "nullptr" << endl;
    }

    void deleteNode(int key){
        Node* temp = head;
        Node* prev = nullptr; 

        if(temp != nullptr && temp->data == key){
            head = temp->next;
            delete temp;
            return;
        }

        while(temp != nullptr && temp->data != key){
            prev = temp;
            temp = temp->next;
        }

        if(temp == nullptr) return;

        prev->next = temp->next;
        delete temp;


    }


};


int main(){
    LinkedList list;

    list.append(1);
    list.append(2);
    list.append(3);
    list.append(4);

    cout << "Linked List:" << endl;
    list.printList();
    cout<< list.head;


    cout << "3 'ü sil:" << endl;
    list.deleteNode(3);
    list.printList();
    return 0;
} 
